### ISSUE TITLE
In-game Options Discussion

### ISSUE THREAD

**Issue by wookieejedi** [scp-fs2open/fs2open.github.com#5697]
Chatting with @MjnMixael about the paradigm for in-game options and wanted to post this here for additional discussion.

Regarding where things are saved for settings, there are some variations:

Game settings that were always in retail (sound level, nebula detail, etc) are saved in the player file, which is great since different mods and pilots can keep settings. Unfortunately, the settings like resolution and all the graphic in-game options that relate to command lines are not saved in the player file but rather the singular common `fs2_open.ini`. 

With in-game options you can open one mod, toggle a graphic option like `soft_particles` on, then whatever mod you run after that will have `soft_particles` enabled (unless that mod has `-disable soft_particles` in its command line). 

Though given that FSO needs to read those graphic options before starting that's why they are not in the player file. Still, it definitely made it a bit puzzling at first why some settings stuck across various mods and others did not.

From Mjn
`I think resolving this will require a kind of fundamental shift to how FreespaceOpen is treated. For decades we've considered "FreespaceOpen" the game and the mods as the game content. But this requires that we consider mods as their own unique games and thus we probably need to look at modifying how data is stored in the .ini again or provide some other new file to check for in-game options as overrides to everything else if they exist.`

Posting this as an issue here to promote additional discussion in the community.

**Comment by Baezon** [comment]
It's not entirely clear what the actual problem is here to me? Some options persist between mods and others do not, is that right? Unusual, and perhaps unideal certainly, but is this actually a significant issues or a mod or the player?

**Comment by MjnMixael** [comment]
I think so. Take an option like Deferred Lighting, for example. My laptop can a simple mod with that on with very little issue. Switch to BtA and suddenly I need to turn Deferred Lighting off just to get the game to run at a playable framerate. Ideally that change would only apply to BtA and not the other mod.

Take another example of mod customization. Wookie and I talked about making the preset levels for these options (low, medium, high, ultra) configurable in SCPUI's options.tbl so a particular mod can fine-tune how their mods run for players based on the content they provide. As-is that's kind of a moot idea because BtA's settings will conflict with any other mod's settings.

Essentially, many of these options really depend on the type of content that is provided when the engine is being played so ideally these options would be saved based on that same definition.

**Comment by wookieejedi** [comment]
Good question. What was most unideal is that all mods that use in game options share the same ini file, so changing a setting in one mod will affect any others. A possible option would be to have mod specific ini files I suppose, similar to how Knossos makes mod specific command line jsons

**Comment by MjnMixael** [comment]
Another thought I just had is that from the player's perspective this is essentially a digression. Since at least wxLauncher we've saved these option selections for the player on a per-mod basis. What in-game options does is moving handling of these things from the launcher to the engine itself. We should provide the same functionality.
